DependencyInjection/Compiler/TokenStorageCompilerPass.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the FOSOAuthServerBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\OAuthServerBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @author Andras Ratz <ratz.andras86@gmail.com>
+ */
+class TokenStorageCompilerPass implements CompilerPassInterface
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $definition = $container->getDefinition('fos_oauth_server.security.authentication.listener');
+
+        if ($container->hasDefinition('security.token_storage')) {
+            $tokenStorageReference = new Reference('security.token_storage');
```

Comment:
no, it won't throw an error in 2.3 as your compiler pass will replace the argument here *before* the reference graph is validated.

My idea is to make the XML file valid already for newest version of Symfony, and then having the compiler pass replace the argument for Symfony 2.3 only. This way, dropping support for Symfony 2.3 means we can simply remove the compiler pass, without having to change the XML file again (while your current proposal requires the compiler pass to change the config for any version of Symfony, so removing it without changing the XML file would break the config)